src/Common/RemoteProxyConfigurationResolver.cpp

```diff
@@ -16,22 +15,41 @@ namespace ErrorCodes
     extern const int BAD_ARGUMENTS;
 }
 
+std::string RemoteProxyHostFetcherImpl::fetch(const Poco::URI & endpoint, const ConnectionTimeouts & timeouts)
+{
+    auto request = Poco::Net::HTTPRequest(Poco::Net::HTTPRequest::HTTP_GET, endpoint.getPath(), Poco::Net::HTTPRequest::HTTP_1_1);
+    auto session = makeHTTPSession(HTTPConnectionGroupType::HTTP, endpoint, timeouts);
+
+    session->sendRequest(request);
+
+    Poco::Net::HTTPResponse response;
+    auto & response_body_stream = session->receiveResponse(response);
+
+    if (response.getStatus() != Poco::Net::HTTPResponse::HTTP_OK)
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Proxy resolver returned not OK status: {}", response.getReason());
```

Comment:
`BAD_ARGUMENTS` do not understand here that error. That error refer to the user mistake, but that is out of his control.
` throw HTTPException(code, uri, status, response.getReason(), body.str());`

Reply:
Is the `body` mandatory? Followed: https://github.com/ClickHouse/ClickHouse/blob/a49f5e43437f03b75a94a1af702a1862396dd568/src/IO/ReadWriteBufferFromHTTP.cpp#L405

Reply:
> Is the body mandatory?
No